crates/mun_memory/src/diff.rs

```diff
@@ -74,40 +78,21 @@ impl PartialOrd for Diff {
 /// Given an `old` and a `new` set of types, calculates the difference.
```

Comment:
Could you elaborate a little more on this comment?

Something like:

```suggestion
/// Given an `old` and a `new` set of types, calculate the operations required to convert the `old` set to the `new` set.
```

I also think the name could be more descriptive, e.g.: `compute_diff`?